aqt/installer.py

```diff
@@ -824,21 +824,38 @@ def run_installer(archives: List[QtPackage], base_dir: str, sevenzip: Optional[s
         tasks.append((arc, base_dir, sevenzip, queue, keep))
     ctx = multiprocessing.get_context("spawn")
     pool = ctx.Pool(Settings.concurrency, init_worker_sh)
+
+    def close_worker_pool_on_exception(exception: BaseException):
+        logger = getLogger("aqt.installer")
+        logger.warning(f"Caught {exception.__class__.__name__}, terminating installer workers")
+        pool.terminate()
+        pool.join()
```

Comment:
It may be possible to move this block into the `finally` clause, so we don't have to call it in each `except` clause. I'm not sure it's worth doing though.

Reply:
No need to move. Current is better to maintain.